feat: prune origin during switch

mcwarman opened the pull request:
## Motivation
Remove the gone remotes during switch back to default after the merge.

## Changes

<!-- Bits between these two tags can be used as the squash_merge_commit_message when the PR is
     approved and merged -->
<!-- SQUASH_MERGE_START -->
- add remote prune before delete
- switch using switch over checkout
<!-- SQUASH_MERGE_END -->

mcwarman commented on bb-pr:
```diff
@@ -319,8 +319,9 @@ action_checkout() {
-    git checkout "$GIT_REMOTE_DEFAULT"
+    git switch "$GIT_REMOTE_DEFAULT"
     git pull
+    git remote prune origin
```
Maybe prune isn't the correct answer:

```
URL: git@bitbucket.org:....git
 * [pruned] origin/feat/...
 * [pruned] origin/feat/....
 * [pruned] origin/feat/...
 * [pruned] origin/feat/...
 * [pruned] origin/feature/...
Deleted branch feat/... (was xxxx).
````

quotidian-ennui replied:
Might be best to leave it as an exercise for the user to do a `git fetch origin --prune --prune-tags`

It'll be that 1% edge case where you really burn yourself.

mcwarman replied:
@quotidian-ennui What about:

```shell
git branch --remotes --delete $(git rev-parse --abbrev-ref --symbolic-full-name @{u})
```

mcwarman commented:
@quotidian-ennui fix worked as expected:

```
Deleted remote-tracking branch origin/refactor/.... (was xxx).
Deleted branch refactor/.... (was xxx).
```
